test/test_grid.py

```diff
@@ -534,9 +509,225 @@ def test_edge_nodes_euler(self):
             grid_xr = xr.open_dataset(grid_path)
             grid_ux = ux.Grid(grid_xr)
 
-            n_face = grid_ux.nMesh2_node
-            n_node = grid_ux.nMesh2_face
+            n_face = grid_ux.nMesh2_face
+            n_node = grid_ux.nMesh2_node
             n_edge = grid_ux.nMesh2_edge
 
             # euler's formula (n_face = n_edges - n_nodes + 2)
             assert (n_face == n_edge - n_node + 2)
+
+    def test_build_face_edges_connectivity(self):
+        """Generates Grid.Mesh2_edge_nodes from Grid.Mesh2_face_nodes."""
```

Comment:
A more detailed docstring would be good considering the length of this test case